### PR TITLE
Add documentation for Bryant Evolution integration.

### PR DESCRIPTION
**Pull request by danielsmyers**
## Proposed change

Add documentation for Bryant Evolution integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/119788
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/5714
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
  - Introduced comprehensive integration instructions for Bryant Evolution HVAC systems using a System Access Module in Home Assistant.
  - Included details on required components, installation steps, and compatibility with HomeKit and local polling.

- **Chores**
  - Updated `.gitignore` to exclude the `Gemfile.lock` file from version control for improved dependency management.

### CODE REVIEW

**Comment by joostlek** [comment]
@danielsmyers seems like there is no brands PR open yet, can you look into that?

File: Gemfile.lock [hunk 1]
```diff
@@ -126,8 +126,9 @@ GEM
     ruby2_keywords (0.0.5)
     safe_yaml (1.0.5)
     sass (3.4.25)
-    sass-embedded (1.77.2-x86_64-linux-gnu)
+    sass-embedded (1.77.5)
```
Comment (c0ffeeca7) [inline Gemfile.lock]:
```suggestion
    sass-embedded (1.77.2-x86_64-linux-gnu)
```

File: Gemfile.lock [hunk 2]
```diff
@@ -126,8 +126,9 @@ GEM
...
       google-protobuf (>= 3.25, < 5.0)
+      rake (>= 13)
```
Comment (c0ffeeca7) [inline Gemfile.lock]:
```suggestion
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]
Undo Gemfile changes. They are unrelated to this PR.

File: source/_integrations/bryant_evolution.markdown [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
+title: Bryant Evolution
+description: Instructions for how to integrate a Bryant Evolution HVAC system using a System Access Module.
+ha_category:
+  - Climate
+featured: false
+ha_release: 2024.???
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]:
```suggestion
ha_release: 2024.8
```

File: source/_integrations/bryant_evolution.markdown [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ha_domain: bryant_evolution
+ha_platforms:
+  - climate
+ha_zeroconf: false
+ha_homekit: true
+ha_integration_type: integration
+---
+
+The **Bryant Evolution** {% term integration %} lets you control Bryant Evolution HVAC systems.
+
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown:19]:
```suggestion

## Prerequisites

```

File: source/_integrations/bryant_evolution.markdown [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+It requires that a System Access Module be connected to the HVAC system (e.g., a Bryant SYSTXBBRCT01).
+
+To install:
+
+1. Connect the System Access Module to a computer using an RS-232-to-USB adapter.
+1. Download [evolutionhttp](https://github.com/danielsmyers/evolutionhttp). Build and run the server binary, passing --device=/dev/ttyUSB0 (or whatever device your adapter is connected to).
+1. Install the bryant_evolution integration in Home Assistant, passing the IP or hostname of the computer to which the System Access Module is connected.
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]:
```suggestion
1. Install the bryant_evolution integration in Home Assistant, passing the IP or hostname of the computer to which the System Access Module is connected.

{% include integrations/config_flow.md %}

```

File: source/_integrations/bryant_evolution.markdown [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ha_platforms:
+  - climate
+ha_zeroconf: false
+ha_homekit: true
+ha_integration_type: integration
+---
+
+The **Bryant Evolution** {% term integration %} lets you control Bryant Evolution HVAC systems.
+
+It requires that a System Access Module be connected to the HVAC system (e.g., a Bryant SYSTXBBRCT01).
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]:
```suggestion
 - It requires that a System Access Module be connected to the HVAC system (for example, a Bryant SYSTXBBRCT01).
```

File: source/_integrations/bryant_evolution.markdown [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+To install:
+
+1. Connect the System Access Module to a computer using an RS-232-to-USB adapter.
+1. Download [evolutionhttp](https://github.com/danielsmyers/evolutionhttp). Build and run the server binary, passing --device=/dev/ttyUSB0 (or whatever device your adapter is connected to).
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]:
```suggestion
2. Download [evolutionhttp](https://github.com/danielsmyers/evolutionhttp). 
3. Build and run the server binary, passing --device=/dev/ttyUSB0 (or whatever device your adapter is connected to).
```

File: source/_integrations/bryant_evolution.markdown (hunk 5 above)
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]:
```suggestion
4. Install the bryant_evolution integration in Home Assistant, passing the IP or hostname of the computer to which the System Access Module is connected.
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]

**Comment by danielsmyers** [comment]
> @danielsmyers seems like there is no brands PR open yet, can you look into that?

https://github.com/home-assistant/brands/pull/5714

File: .gitignore [hunk 8]
```diff
@@ -22,3 +22,4 @@ source/.jekyll-metadata
 !.vscode/tasks.json
 *.suo
 tags*
+Gemfile.lock
```
Comment (c0ffeeca7) [inline .gitignore]: Please only include files that are relevant to this PR. Don't push gitignore or Gemfile.lock.

File: source/_integrations/bryant_evolution.markdown [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+The **Bryant Evolution** {% term integration %} lets you control Bryant Evolution HVAC systems.
+
+## Prerequisites
+
+- It requires that a System Access Module be connected to the HVAC system (for example, a Bryant SYSTXBBRCT01).
+
+To install:
+
+1. Connect the System Access Module to the device running Home Assistant, e.g., using an RS-232-to-USB adapter.
+2. Install the bryant_evolution integration in Home Assistant, passing the name of the serial port to which the device is connected (e.g., `/dev/ttyUSB0`)
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]: in a previous review, I added the config_flow below here. Is there a particular reason this was removed? Does it not apply?
{% include integrations/config_flow.md %}

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]

File: source/_integrations/bryant_evolution.markdown [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+The **Bryant Evolution** {% term integration %} lets you control Bryant Evolution HVAC systems.
+
+## Prerequisites
+
+- It requires that a System Access Module be connected to the HVAC system (for example, a Bryant SYSTXBBRCT01).
+
+To install:
+
+1. Connect the System Access Module to the device running Home Assistant, e.g., using an RS-232-to-USB adapter.
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]:
```suggestion
1. Connect the System Access Module to the device running Home Assistant, for example, by using an RS-232-to-USB adapter.
```

File: source/_integrations/bryant_evolution.markdown [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+2. Install the bryant_evolution integration in Home Assistant, passing the name of the serial port to which the device is connected (e.g., `/dev/ttyUSB0`)
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown]:
```suggestion
2. Install the Bryant Evolution integration in Home Assistant, passing the name of the serial port to which the device is connected (for example, `/dev/ttyUSB0`)
```

File: source/_integrations/bryant_evolution.markdown [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+## Prerequisites
+
+- It requires that a System Access Module be connected to the HVAC system (for example, a Bryant SYSTXBBRCT01).
+
+To install:
+
+1. Connect the System Access Module to the device running Home Assistant, for example, by using an RS-232-to-USB adapter.
+2. Install the Bryant Evolution integration in Home Assistant, passing the name of the serial port to which the device is connected (for example, `/dev/ttyUSB0`)
+
+{% include integrations/config_flow.md %}
```
Comment (c0ffeeca7) [inline source/_integrations/bryant_evolution.markdown:29]: Below here, please add a section called Climate, providing a bit of detail what this device can do. For example, like the Gree Climate integration docs. 
If there are any supported models, please add a section listing those.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @danielsmyers :+1:

**Pull request merged**